src/lib/yearn-affiliate-wrapper.sol

```diff
@@ -0,0 +1,357 @@
+    uint256 constant DEPOSIT_EVERYTHING = uint256(-1);
+    uint256 constant WITHDRAW_EVERYTHING = uint256(-1);
+    uint256 constant MIGRATE_EVERYTHING = uint256(-1);
+    // VaultsAPI.depositLimit is unlimited
+    uint256 constant UNCAPPED_DEPOSITS = uint256(-1);
+
+    constructor(address _token, address _registry) public {
+        token = IERC20(_token);
+        // v2.registry.ychad.eth
+        registry = RegistryAPI(_registry);
+    }
+
+    function setRegistry(address _registry) external {
+        require(msg.sender == registry.governance());
+        // In case you want to override the registry instead of re-deploying
+        registry = RegistryAPI(_registry);
+    }
+
+    function bestVault() public virtual view returns (VaultAPI) {
+        return VaultAPI(registry.latestVault(address(token)));
+    }
+
+    function allVaults() public virtual view returns (VaultAPI[] memory) {
+        uint256 cache_length = _cachedVaults.length;
+        uint256 num_vaults = registry.numVaults(address(token));
+
+        // Use cached
+        if (cache_length == num_vaults) {
+            return _cachedVaults;
+        }
+
+        VaultAPI[] memory vaults = new VaultAPI[](num_vaults);
+
+        for (uint256 vault_id = 0; vault_id < cache_length; vault_id++) {
+            vaults[vault_id] = _cachedVaults[vault_id];
+        }
+
+        for (uint256 vault_id = cache_length; vault_id < num_vaults; vault_id++) {
+            vaults[vault_id] = VaultAPI(registry.vaults(address(token), vault_id));
+        }
+
+        return vaults;
+    }
+
+    function _updateVaultCache(VaultAPI[] memory vaults) internal {
+        if (vaults.length > _cachedVaults.length) {
+            _cachedVaults = vaults;
+        }
+    }
+
+    function totalVaultBalance(address account) public view returns (uint256 balance) {
+        VaultAPI[] memory vaults = allVaults();
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            balance = balance.add(vaults[id].balanceOf(account).mul(vaults[id].pricePerShare()).div(10**uint256(vaults[id].decimals())));
+        }
+    }
+
+    function totalAssets() public view returns (uint256 assets) {
+        VaultAPI[] memory vaults = allVaults();
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            assets = assets.add(vaults[id].totalAssets());
+        }
+    }
+
+    function _deposit(
+        address depositor,
+        address receiver,
+        uint256 amount, // if `MAX_UINT256`, just deposit everything
+        bool pullFunds // If true, funds need to be pulled from `depositor` via `transferFrom`
+    ) internal returns (uint256 deposited) {
+        VaultAPI _bestVault = bestVault();
+
+        if (pullFunds) {
+            token.safeTransferFrom(depositor, address(this), amount);
+        }
+
+        if (token.allowance(address(this), address(_bestVault)) < amount) {
+            token.safeApprove(address(_bestVault), UNLIMITED_APPROVAL); // Vaults are trusted
```

Comment:
Here we added this line to avoid issues with some tokens

https://github.com/yearn/yearn-vaults/blob/c818595d1699077d20285c5fdd28ed2a9f564b6c/contracts/BaseWrapper.sol#L165